com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -218,6 +220,49 @@ internal struct AgentParameters
         [FormerlySerializedAs("maxStep")]
         [HideInInspector] public int MaxStep;
 
+        /// <summary>
+        /// Options for controlling how the Agent class is searched for <see cref="ObservableAttribute"/>s.
+        /// </summary>
+        public enum ObservableAttributeHandling
+        {
+            /// <summary>
+            /// All ObservableAttributes on the Agent will be ignored. If there are no
+            /// ObservableAttributes on the Agent, this will result in the fastest
+            /// initialization time.
+            /// </summary>
+            Ignore,
+
+            /// <summary>
+            /// Only members on the declared class will be examined; members that are
+            /// inherited are ignored. This is the default behavior, and a reasonable
+            /// tradeoff between performance and flexibility.
+            /// </summary>
+            /// <remarks>This corresponds to setting the
+            /// [BindingFlags.DeclaredOnly](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.bindingflags?view=netcore-3.1)
+            /// when examining the fields and properties of the Agent class instance.
+            /// </remarks>
+            SkipInherited,
+
+            /// <summary>
+            /// All members on the class will be examined. This can lead to slower
+            /// startup times
+            /// </summary>
+            ExamineAll
+        }
+
+        [HideInInspector, SerializeField]
+        ObservableAttributeHandling m_observableAttributeBehavior = ObservableAttributeHandling.SkipInherited;
```

Comment:
I would argue for the default to be "ignore."

Reply:
Yeah, I was on the fence about this. I think since SkipInherited ignores everything in `Agent`, it doesn't add much overhead, and is a better new user experience if they don't need to change this. 

Is there any (easy) way in serialization callbacks that we could set this to Ignore for old assets but keep it to SkipInherited for new ones?

Reply:
Changed to `Ignore`